pkg/apis/kops/validation/validation.go

```diff
@@ -611,7 +611,7 @@ func validateSubnet(subnetSpec *kops.ClusterSubnetSpec, c *kops.ClusterSpec, fie
 				allErrs = append(allErrs, field.Forbidden(fieldPath.Child("cidr"), fmt.Sprintf("subnet %q cidr %q is not a subnet of the networkCIDR %q%s", subnetSpec.Name, subnetSpec.CIDR, c.Networking.NetworkCIDR, extraMsg)))
 			}
 		}
-		if subnet.Overlap(subnetCIDR, podCIDR) {
+		if subnet.Overlap(subnetCIDR, podCIDR) && c.Networking.AmazonVPC == nil {
```

Comment:
I believe Cilium in ENI mode is in the same situation.

I wonder if the correct behavior for ENI-based CNIs is to leave `podCIDR` unset, like we do for IPv6. CCM is not assigning the pod IPs, the CNI is, deriving it from the ENI.